### PR TITLE
PREAPPS-5189 : Adding documentAction for handling sharing of documents

### PR DESCRIPTION
**Pull request by auminpatel**

### CODE REVIEW

**Review by lavegupta — APPROVED** [review]

**Review by itsmeprashant — APPROVED** [review]

File: src/batch-client/index.ts [hunk 1]
```diff
@@ -314,6 +315,21 @@ export class ZimbraBatchClient {
...
 	};
 
+	public documentActionresponse = (type: ActionType, options: ActionOptions) => {
+		const { ids, id, ...rest } = options;
+
+		return this.jsonRequest({
+			name: type,
+			body: {
+				action: {
+					id: id || (ids || []).join(','),
```
Comment (silentsakky) [inline src/batch-client/index.ts:325]: do we need two separate parameters?
`['single'].join(',') ==> 'single'`

Comment (auminpatel) [reply src/batch-client/index.ts]: that works, 
Done

File: src/batch-client/index.ts [hunk 2]
```diff
@@ -314,6 +315,21 @@ export class ZimbraBatchClient {
...
 	};
 
+	public documentActionresponse = (type: ActionType, options: ActionOptions) => {
+		const { ids, id, ...rest } = options;
+
+		return this.jsonRequest({
+			name: type,
+			body: {
+				action: {
+					id: id || [ids].join(','),
```
Comment (silentsakky) [inline src/batch-client/index.ts:325]: do we need two parameters id and ids?

Comment (auminpatel) [reply src/batch-client/index.ts]: not required

Comment (auminpatel) [reply src/batch-client/index.ts]: as this is only used for documentAction response so i have used only id so removed it

File: src/batch-client/index.ts [hunk 3]
```diff
@@ -306,14 +307,29 @@ export class ZimbraBatchClient {
 			name: type,
 			body: {
 				action: {
-					id: id || (ids || []).join(','),
+					id: id || [ids].join(','),
```
Comment (silentsakky) [inline src/batch-client/index.ts:310]: do we need two params ? id and ids?

Comment (auminpatel) [reply src/batch-client/index.ts:310]: Many of the actions are passing ids or id, this part is been use in msgaction, Folderaction itemaction so I haven't checked those mutation

Comment (auminpatel) [reply src/batch-client/index.ts:310]: Screen Shot 2020-10-06 at 7 03 21 AM
just a example

Comment (silentsakky) [reply src/batch-client/index.ts:310]: ok

**Review by silentsakky — APPROVED** [review]

**Pull request merged**